src/util/WellPlateInternal.tsx

```diff
@@ -24,6 +25,95 @@ interface IWellPlateInternalProps extends IWellPlateCommonProps {
 export const WellPlateInternal: FunctionComponent<IWellPlateInternalProps> = (
   props,
 ) => {
+  const { displayAsGrid = false } = props;
+
+  if (!displayAsGrid) {
+    return <DefaultWellPlateInternal {...props} />;
+  }
+
+  return <GridWellPlateInternal {...props} />;
+};
+
+function GridWellPlateInternal(
+  props: Omit<IWellPlateInternalProps, 'displayAsGrid'>,
+) {
+  const { plate } = props;
+
+  const columnLabels = plate.columnLabels;
+  const rowLabels = plate.rowLabels;
+
+  const cellStyle = {
+    borderStyle: 'solid',
+    borderColor: 'gray',
+    borderWidth: 1,
+  };
+
+  const values: Array<{ index: number; label: ReactNode }> = [];
+  for (let i = 0; i <= rowLabels.length - 1; i++) {
+    values.push({ index: undefined, label: rowLabels[i] });
+    for (let j = 0; j <= columnLabels.length - 1; j++) {
+      const index = plate.getIndex({ row: i, column: j });
+      const label = props.text?.(index);
+
+      values.push({
+        index,
+        label: label === undefined ? plate.getPositionCode(index) : label,
+      });
+    }
+  }
+
+  return (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateColumns: `max-content repeat(${columnLabels.length}, 1fr)`,
+        gridTemplateRows: `repeat(${rowLabels.length}, 1fr)`,
+        userSelect: 'none',
+        textAlign: 'center',
+        ...cellStyle,
+      }}
+    >
+      {[
+        { index: undefined, label: '' },
+        ...columnLabels.map((value) => ({ index: undefined, label: value })),
+        ...values,
+      ].map(({ index, label }) => {
+        if (index === undefined) {
+          return (
+            <div key={index} style={{ ...cellStyle, padding: 5 }}>
```

Comment:
You should use something defined and unique as your `key`.  Don't you have react warnings about your keys?